grails-app/domain/org/pih/warehouse/product/Product.groovy

```diff
@@ -282,7 +282,7 @@ class Product implements Comparable, Serializable {
         productCode(nullable: true, maxSize: 255, unique: true)
         unitOfMeasure(nullable: true, maxSize: 255)
         category(nullable: false)
-        productType(nullable: true)
+        productType(nullable: false)
```

Comment:
Reminder to self: Should see a database migration to set product type for all existing products.